mParticle-Apple-SDK/Utils/MPLaunchInfo.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  MPLaunchInfo.swift
+//  mParticle-Apple-SDK
+//
+//  Created by Brandon Stalnaker on 12/18/24.
+//
+
+import Foundation
+@objc public class MPLaunchInfo : NSObject {
+    @objc private var sourceApp: String?
+    @objc public private(set) var url: URL
+    @objc public private(set) var sourceApplication: String?
+    @objc public private(set) var annotation: String?
+    @objc public private(set) var options: [String : Any]?
+
+    @objc(initWithURL:sourceApplication:annotation:) public init?(url: URL?, sourceApplication: String?, annotation: Any?) {
+        guard let url else { return nil }
```

Comment:
Just a note for discussion later to add to the style guide, but I personally really like single line guard statements like this. I think I've been doing them elsewhere with new lines like this:

```
guard let url else { 
    return nil 
}
```

But I was only doing that because we tended to not use single line statements in other places, but I would love to discuss this and add something official to the style guide about it.